clang/include/clang/Basic/Builtins.td

```diff
@@ -4737,6 +4737,12 @@ def HLSLRSqrt : LangBuiltin<"HLSL_LANG"> {
   let Prototype = "void(...)";
 }
 
+def HLSLSign : LangBuiltin<"HLSL_LANG"> {
+  let Spellings = ["__builtin_hlsl_elementwise_sign"];
+  let Attributes = [NoThrow, Const];
+  let Prototype = "int(...)";
```

Comment:
This is not correct Since we don't use `CustomTypeChecking` int will apply scalar integer rules. Since there is no way to specify int vector and int scalar we ussually just go with void so that no rules are applied.